internal/mw/verbs_allowed.go

```diff
@@ -0,0 +1,22 @@
+package mw
+
+import "net/http"
+
+// VerbsAllowed is a middleware that allows only specific HTTP verbs to be
+// processed. If the request verb is not in the list of allowed verbs, a
+// 405 Method Not Allowed response is returned.
+func VerbsAllowed(allowedVerbs ...string) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			for _, allowedVerb := range allowedVerbs {
+				if r.Method == allowedVerb {
+					next.ServeHTTP(w, r)
+					return
+				}
+			}
+
+			w.WriteHeader(http.StatusMethodNotAllowed)
+			w.Write([]byte("405 method not allowed"))
+		})
+	}
+}
```

Comment:
Small suggestion, although the loop will always be small, given the number of HTTP methods available, it will be executed for each request. What do you think about building a map at the invocation and then just using it to do a lookup in constant time?

```suggestion
func VerbsAllowed(allowedVerbs ...string) func(http.Handler) http.Handler {
	allowed := make(map[string]struct{}, len(allowedVerbs))
	for _, allowedVerb := range allowedVerbs {
		allowed[allowedVerb] = struct{}{}
	}

	return func(next http.Handler) http.Handler {
		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
			if _, ok := allowed[r.Method]; ok {
				next.ServeHTTP(w, r)
				return
			}

			w.WriteHeader(http.StatusMethodNotAllowed)
			w.Write([]byte("405 method not allowed"))
		})
	}
}
```

Reply:
For the time being, I'll keep it as a slice for loop. It _should_ be better at handling this _faster_ than the map.